peachjam/views/save_document.py

```diff
@@ -0,0 +1,153 @@
+from django.contrib.auth import get_user_model
+from django.contrib.auth.mixins import LoginRequiredMixin, PermissionRequiredMixin
+from django.http import Http404, HttpResponse, HttpResponseRedirect
+from django.shortcuts import get_object_or_404
+from django.urls import reverse, reverse_lazy
+from django.views.generic import (
+    CreateView,
+    DeleteView,
+    ListView,
+    TemplateView,
+    UpdateView,
+)
+from django.views.generic.edit import FormMixin
+
+from peachjam.forms import FolderForm, SaveDocumentForm
+from peachjam.models import CoreDocument, Folder, SavedDocument, pj_settings
+
+User = get_user_model()
+
+
+class AllowSavedDocumentMixin:
+    def dispatch(self, *args, **kwargs):
+        if not pj_settings().allow_save_documents:
+
+            raise Http404("Saving documents is not allowed.")
+        return super().dispatch(*args, **kwargs)
+
+
+class BaseFolderView(
```

Comment:
Best practice is to name this `...Mixin` so that it's clear it's not a concrete view.

Reply:
done